typer.ml

```diff
@@ -3504,6 +3504,27 @@ and type_expr ctx (e,p) (with_type:with_type) =
 		let e = type_expr ctx e (WithType t) in
 		let e = Codegen.AbstractCast.cast_or_unify ctx t e p in
 		if e.etype == t then e else mk (TCast (e,None)) t p
+	| EMeta ((Meta.StoredTypedExpr,_,_),(EConst (Int s),_)) ->
+		let vars = Hashtbl.create 0 in
+		let rec build_expr e =
+			match e.eexpr with
+			| TVar (v,eo) ->
```

Comment:
About the EMeta case - I don't think so. It does the whole typing and then checks for meta, while we should extract the id and simply return a typed id from the PMap.